aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -116,9 +129,14 @@ Tensor empty_like(const Tensor& self) {
 
 Tensor empty_like(const Tensor& self, const TensorOptions& options) {
   if (options.layout() == kSparse && self.type().is_sparse()) {
+#ifndef USE_TH_SIZE_ZERO_DIM
     auto res = options.type().tensor({});
```

Comment:
maybe related to the above, but shouldn't this return a scalar tensor?  You presumably want `tensor()` or `tensor({0})` here.

Reply:
Oh, I guess this is okay because it will be ifdef'ed out.  Sorry!

Reply:
I should rename `sparse_raw_resize_` to `sparse_raw_resize_legacy_` for better readability.